In [107]:
#-*- coding: utf-8 -*-
import pandas as pd
import os
import urllib
import datetime
import time
import random
import numpy as np
import copy

In [5]:
companyListPath = "/Users/moon/1save/JpStock/CompanyNameList"

fileList = os.listdir(companyListPath)

print(fileList[-1]) #這個好像會照順序排，那你直接取 -1就可以了

CompanyNameList_2017-06-19.csv


In [34]:
companyPd = pd.read_csv(os.path.join(companyListPath, fileList[-1]), names = ["StockSymbol"])

#print(companyPd[:5])

for each in companyPd["StockSymbol"][:3]:
    print(each)

1301-T
1305-T
1306-T


In [77]:
len(companyPd)

4026

In [16]:
dailyDFhtml = "http://k-db.com/stocks/"

daily5min = "http://k-db.com/stocks/1301-T/5m?download=csv"

test = pd.read_csv(daily5min)

In [17]:
test

���t   ����    �n�l    ���l    ���l    �I�l  �o����   �������
0   2017-06-19  15:00  3150.0  3150.0  3150.0  3150.0    2000   6300000
1   2017-06-19  14:55  3155.0  3160.0  3150.0  3160.0    1800   5678500
2   2017-06-19  14:50  3155.0  3160.0  3155.0  3160.0     200    631500
3   2017-06-19  14:45  3155.0  3155.0  3155.0  3155.0     200    631000
4   2017-06-19  14:40  3160.0  3160.0  3155.0  3155.0     700   2211500
5   2017-06-19  14:35  3160.0  3160.0  3160.0  3160.0     100    316000
6   2017-06-19  14:30  3160.0  3160.0  3160.0  3160.0     400   1264000
7   2017-06-19  14:25  3160.0  3160.0  3160.0  3160.0     400   1264000
8   2017-06-19  14:20  3155.0  3155.0  3155.0  3155.0     100    315500
9   2017-06-19  14:15     NaN     NaN     NaN     NaN       0         0
10  2017-06-19  14:10  3155.0  3155.0  3155.0  3155.0     100    315500
11  2017-06-19  14:05  3155.0  3155.0  3155.0  3155.0     500   1577500
12  2017-06-19  14:00  3160.0  3160.0  3155.0  3155.0    1400   4422000
13  2017-06-19  13:55  3165.0  3165.0  3160.0  3160.0    1400   4429500
14  2017-06-19  13:50     NaN     NaN     NaN     NaN       0         0
15  2017-06-19  13:45  3165.0  3165.0  3165.0  3165.0     500   1582500
16  2017-06-19  13:40  3165.0  3165.0  3165.0  3165.0     900   2848500
17  2017-06-19  13:35     NaN     NaN     NaN     NaN       0         0
18  2017-06-19  13:30  3165.0  3165.0  3165.0  3165.0     400   1266000
19  2017-06-19  13:25  3165.0  3165.0  3165.0  3165.0     200    633000
20  2017-06-19  13:20  3170.0  3170.0  3165.0  3170.0    1100   3483000
21  2017-06-19  13:15  3160.0  3175.0  3160.0  3175.0    2000   6340500
22  2017-06-19  13:10  3165.0  3165.0  3160.0  3160.0     200    632500
23  2017-06-19  13:05  3165.0  3165.0  3165.0  3165.0     800   2532000
24  2017-06-19  13:00  3160.0  3165.0  3160.0  3165.0    2200   6954000
25  2017-06-19  12:55  3155.0  3160.0  3155.0  3160.0     200    631500
26  2017-06-19  12:50  3155.0  3155.0  3155.0  3155.0     100    315500
27  2017-06-19  12:45  3155.0  3155.0  3155.0  3155.0     300    946500
28  2017-06-19  12:40  3155.0  3155.0  3155.0  3155.0     200    631000
29  2017-06-19  12:35  3155.0  3155.0  3155.0  3155.0     100    315500
..         ...    ...     ...     ...     ...     ...     ...       ...
43  2017-06-19  11:25     NaN     NaN     NaN     NaN       0         0
44  2017-06-19  11:20  3155.0  3155.0  3155.0  3155.0     200    631000
45  2017-06-19  11:15  3155.0  3155.0  3155.0  3155.0     500   1577500
46  2017-06-19  11:10  3150.0  3155.0  3150.0  3155.0    2100   6621500
47  2017-06-19  11:05  3150.0  3150.0  3150.0  3150.0     100    315000
48  2017-06-19  11:00  3155.0  3155.0  3150.0  3150.0     300    945500
49  2017-06-19  10:55     NaN     NaN     NaN     NaN       0         0
50  2017-06-19  10:50  3150.0  3150.0  3150.0  3150.0     100    315000
51  2017-06-19  10:45  3150.0  3150.0  3150.0  3150.0     300    945000
52  2017-06-19  10:40  3145.0  3150.0  3145.0  3150.0    2200   6921000
53  2017-06-19  10:35     NaN     NaN     NaN     NaN       0         0
54  2017-06-19  10:30     NaN     NaN     NaN     NaN       0         0
55  2017-06-19  10:25     NaN     NaN     NaN     NaN       0         0
56  2017-06-19  10:20  3150.0  3150.0  3150.0  3150.0     900   2835000
57  2017-06-19  10:15     NaN     NaN     NaN     NaN       0         0
58  2017-06-19  10:10  3155.0  3155.0  3155.0  3155.0     400   1262000
59  2017-06-19  10:05     NaN     NaN     NaN     NaN       0         0
60  2017-06-19  10:00     NaN     NaN     NaN     NaN       0         0
61  2017-06-19  09:55  3155.0  3155.0  3155.0  3155.0     700   2208500
62  2017-06-19  09:50  3155.0  3155.0  3155.0  3155.0     300    946500
63  2017-06-19  09:45  3155.0  3155.0  3155.0  3155.0    1200   3786000
64  2017-06-19  09:40  3155.0  3155.0  3155.0  3155.0     900   2839500
65  2017-06-19  09:35  3155.0  3155.0  3155.0  3155.0     600   1893000
66  2017-06-19  09:30  3150.0  3150.0  3150.0  315

In [90]:
daily5min = "http://k-db.com/stocks/1301-T/5m?download=csv"
todayStr = datetime.datetime.today().strftime('%Y-%m-%d') #因為時間已過，所以先寫死
#todayStr = "2017-06-19"
testfile = urllib.URLopener()
testfile.retrieve(daily5min, "./" + todayStr + "/1301-T.csv")

('./2017-06-20/1301-T.csv', <httplib.HTTPMessage instance at 0x10eed36c8>)

In [59]:
for eachS in companyPd["StockSymbol"]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(2,5)
    time.sleep(sleepTime)

1301-T was done
1305-T was done
1306-T was done
1308-T was done
1309-T was done
1310-T was done
1311-T was done
1312-T was done
1313-T was done
1314-T was done
1319-T was done
1319-S was done
1319-F was done
1320-T was done
1321-T was done
1322-T was done
1323-T was done
1324-T was done
1325-T was done
1326-T was done
1327-T was done
1328-T was done
1329-T was done
1330-T was done
1332-T was done
1333-T was done
1343-T was done
1344-T was done
1345-T was done
1346-T was done
1347-T was done
1348-T was done
1349-T was done
1352-T was done
1356-T was done
1357-T was done
1358-T was done
1360-T was done
1361-T was done
1362-T was done
1363-T was done
1364-T was done
1365-T was done
1366-T was done
1367-T was done
1368-T was done
1369-T was done
1376-T was done
1377-T was done
1379-T was done
1380-T was done
1381-T was done
1382-T was done
1383-T was done
1384-T was done
1385-T was done
1386-T was done
1387-T was done
1388-T was done
1389-T was done
1390-T was done
1391-T was done
1392-T w

IOError: ('http error', 503, 'Service Unavailable', <httplib.HTTPMessage instance at 0x112caa5f0>)

In [57]:
print(random.randint(2,5))

4


#### 開始時間，大約 am 1:00
#### 結束時間，未定，等結束後紀錄

## 失敗了！他1560-T 有抓完，但出現了 503 error

In [61]:
print(np.where(companyPd["StockSymbol"]=="1560-T"))

(array([147]),)


In [67]:
print(companyPd["StockSymbol"][147])
print(companyPd["StockSymbol"][148])

1560-T
1561-T


In [68]:
daily5min = "http://k-db.com/stocks/1561-T/5m?download=csv"
#todayStr = datetime.datetime.today().strftime('%Y-%m-%d') 因為時間已過，所以先寫死
todayStr = "2017-06-19"
testfile = urllib.URLopener()
testfile.retrieve(daily5min, "./" + todayStr + "/1561-T.csv")

('./2017-06-19/1561-T.csv', <httplib.HTTPMessage instance at 0x112caa320>)

In [70]:
#print(companyPd["StockSymbol"][149:152])
for eachS in companyPd["StockSymbol"][149:]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)


1563-T was done
1566-T was done
1567-T was done
1568-T was done
1569-T was done
1570-T was done
1571-T was done
1572-T was done
1573-T was done
1574-T was done
1575-T was done
1576-T was done
1577-T was done
1578-T was done
1579-T was done
1580-T was done
1581-T was done
1582-T was done
1583-T was done
1584-T was done
1585-T was done
1586-T was done
1587-T was done
1588-T was done
1589-T was done
1590-T was done
1591-T was done
1592-T was done
1593-T was done
1595-T was done
1596-T was done
1597-T was done
1598-T was done
1599-T was done
1605-T was done
1606-T was done
1610-T was done
1612-T was done
1613-T was done
1615-T was done
1617-T was done
1618-T was done
1619-T was done
1620-T was done
1621-T was done
1622-T was done
1623-T was done
1624-T was done
1625-T was done
1626-T was done
1627-T was done
1628-T was done
1629-T was done
1630-T was done
1631-T was done
1632-T was done
1633-T was done
1634-T was done
1635-T was done
1636-T was done
1637-T was done
1638-T was done
1639-T w

IOError: ('http error', 503, 'Service Unavailable', <httplib.HTTPMessage instance at 0x112c7cdd0>)

In [71]:
print(np.where(companyPd["StockSymbol"]=="1677-T"))

(array([232]),)


發現他停在 232

232-149 = 83

一個批次抓50檔好了

In [72]:
for eachS in companyPd["StockSymbol"][233:250]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1678-T was done
1679-T was done
1680-T was done
1681-T was done
1682-T was done
1683-T was done
1684-T was done
1685-T was done
1686-T was done
1687-T was done
1688-T was done
1689-T was done
1690-T was done
1691-T was done
1692-T was done
1693-T was done
1694-T was done


In [73]:
for eachS in companyPd["StockSymbol"][251:300]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1696-T was done
1697-T was done
1698-T was done
1699-T was done
1711-T was done
1712-T was done
1716-T was done
1717-T was done
1718-T was done
1719-T was done
1720-T was done
1721-T was done
1722-T was done
1723-T was done
1724-T was done
1726-T was done
1728-T was done
1730-T was done
1734-S was done
1736-T was done
1737-T was done
1739-T was done
1743-T was done
1743-F was done
1757-T was done
1758-T was done
1762-T was done
1764-T was done
1766-T was done
1768-T was done
1770-T was done
1771-F was done
1773-T was done
1775-T was done


IOError: ('http error', 503, 'Service Unavailable', <httplib.HTTPMessage instance at 0x10eedd758>)

In [75]:
print(np.where(companyPd["StockSymbol"]=="1775-T"))

(array([284]),)


In [76]:
for eachS in companyPd["StockSymbol"][285:300]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1776-T was done
1780-T was done
1782-T was done
1783-T was done
1787-T was done
1788-T was done
1789-T was done
1793-T was done
1795-T was done
1798-T was done
1799-T was done
1801-T was done
1802-T was done
1802-F was done
1803-T was done


In [78]:
for eachS in companyPd["StockSymbol"][301:330]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1807-T was done
1808-T was done
1810-T was done
1811-T was done
1812-T was done
1813-T was done
1814-T was done
1815-T was done
1820-T was done
1821-T was done
1822-T was done
1824-T was done
1826-T was done
1827-T was done


IOError: ('http error', 503, 'Service Unavailable', <httplib.HTTPMessage instance at 0x10eeddb00>)

In [80]:
print(np.where(companyPd["StockSymbol"]=="1827-T"))

(array([314]),)


In [81]:
for eachS in companyPd["StockSymbol"][315:330]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(1)
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1828-T was done
1832-S was done
1833-T was done
1835-T was done
1840-T was done
1840-S was done
1841-T was done
1844-T was done
1846-T was done
1847-T was done
1848-T was done
1848-F was done


IOError: ('http error', 503, 'Service Unavailable', <httplib.HTTPMessage instance at 0x10eedd518>)

In [82]:
print(np.where(companyPd["StockSymbol"]=="1848-F"))

(array([326]),)


In [83]:
for eachS in companyPd["StockSymbol"][327:330]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(1)
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1850-T was done
1852-T was done
1853-T was done


In [84]:
for eachS in companyPd["StockSymbol"][331:350]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(1)
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1861-T was done
1865-T was done
1866-T was done
1867-T was done
1868-T was done
1870-T was done
1871-T was done
1873-T was done
1878-T was done
1879-T was done
1881-T was done
1881-S was done
1882-T was done
1883-T was done
1884-T was done
1885-T was done
1885-S was done
1888-T was done
1890-T was done


In [85]:
for eachS in companyPd["StockSymbol"][351:380]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(1)
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1896-T was done
1897-T was done
1898-T was done
1899-T was done


IOError: ('http error', 503, 'Service Unavailable', <httplib.HTTPMessage instance at 0x10eed3128>)

In [86]:
for eachS in companyPd["StockSymbol"][351:380]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(2)
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1896-T was done
1897-T was done
1898-T was done
1899-T was done
1904-T was done
1905-T was done
1906-T was done
1909-T was done
1911-T was done
1914-T was done
1916-T was done
1919-T was done
1921-T was done
1921-S was done
1924-T was done
1925-T was done
1926-T was done
1928-T was done
1929-T was done
1930-T was done
1931-T was done
1933-T was done
1933-F was done
1934-T was done
1935-T was done
1937-T was done
1937-F was done
1938-T was done
1939-T was done


In [88]:
for eachS in companyPd["StockSymbol"][381:400]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(2)
    eachFile.retrieve(tempURL, eachPath)
    print(eachS + " was done")
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1942-T was done
1944-T was done
1945-T was done
1946-T was done
1948-T was done
1949-T was done
1950-T was done
1951-T was done
1952-T was done
1954-T was done
1956-T was done
1959-T was done
1959-F was done
1960-T was done
1961-T was done
1963-T was done
1964-T was done
1965-T was done
1966-T was done


In [93]:
todayStr = "2017-06-19"
print(todayStr)

2017-06-19


In [95]:
errList = []

for eachS in companyPd["StockSymbol"][401:500]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m/20170619?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(2)
    try:
        eachFile.retrieve(tempURL, eachPath)
    except IOError:
        errList.append(eachS)
        print(eachS + " got IOError, added to errList")
    else:
        print(eachS + " was done")
    
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

1967-T was done
1968-T was done
1969-T was done
1971-T was done
1972-T was done
1973-T was done
1975-T was done
1976-T was done
1979-T was done
1980-T was done
1981-T was done
1982-T was done
1983-T was done
1984-T got IOError, added to errList
1987-T was done
1989-T got IOError, added to errList
1992-T was done
1994-T got IOError, added to errList
1997-T was done
1999-F was done
2001-T was done
2002-T was done
2003-T was done
2004-T was done
2008-T was done
2009-T was done
2009-F was done
2031-T was done
2032-T was done
2033-T was done
2034-T was done
2035-T was done
2036-T was done
2037-T was done
2038-T was done
2039-T was done
2040-T was done
2041-T was done
2042-T was done
2043-T was done
2044-T was done
2045-T was done
2046-T was done
2047-T was done
2048-T was done
2049-T was done
2053-T was done
2055-T was done
2058-F was done
2060-T was done
2065-T was done
2066-T was done
2067-T was done
2068-T was done
2069-T was done
2107-T was done
2108-T was done
2109-T was done
2112-T wa

In [96]:
print (errList)

['1984-T', '1989-T', '1994-T', '2164-T']


In [97]:
for eachS in companyPd["StockSymbol"][501:]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m/20170619?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(2)
    try:
        eachFile.retrieve(tempURL, eachPath)
    except IOError:
        errList.append(eachS)
        print(eachS + " got IOError, added to errList")
    else:
        print(eachS + " was done")
    
    sleepTime = random.randint(1,4)
    time.sleep(sleepTime)

2181-T was done
2183-T was done
2185-T was done
2186-T was done
2191-T was done
2193-T was done
2195-T was done
2196-T was done
2198-T was done
2201-T was done
2204-T was done
2206-T was done
2207-T was done
2208-T was done
2209-T was done
2211-T was done
2212-T was done
2215-T was done
2216-T was done
2217-T was done
2218-S was done
2220-T was done
2221-T was done
2222-T was done
2224-T was done
2226-T was done
2228-T was done
2229-T was done
2230-T was done
2264-T was done
2266-T was done
2267-T was done
2268-T was done
2269-T was done
2270-T was done
2270-S was done
2281-T was done
2282-T was done
2286-T was done
2288-T was done
2291-T was done
2292-T was done
2293-T was done
2294-T was done
2296-T was done
2300-T was done
2301-T was done
2303-T was done
2304-T was done
2305-T was done
2307-T was done
2309-T was done
2311-T was done
2315-T was done
2317-T was done
2321-T was done
2323-T was done
2325-T was done
2326-T was done
2327-T was done
2329-T was done
2330-T was done
2331-T w

3286-T was done
3286-F was done
3287-T was done
3288-T was done
3289-T was done
3290-T was done
3291-T was done
3292-T was done
3293-T was done
3294-T was done
3295-T was done
3296-T was done
3297-T was done
3297-F was done
3298-T was done
3299-T was done
3300-T was done
3302-T was done
3306-T was done
3308-T was done
3309-T was done
3313-T was done
3315-T was done
3316-T was done
3317-T was done
3318-T was done
3319-T was done
3320-T was done
3321-T was done
3322-T was done
3323-T was done
3326-T was done
3328-T was done
3329-T was done
3333-T was done
3341-T was done
3344-T was done
3347-T was done
3349-T was done
3350-T was done
3352-T was done
3353-T was done
3355-T was done
3356-T was done
3358-T was done
3359-T was done
3359-F was done
3360-T was done
3361-T was done
3366-T was done
3370-T was done
3371-T was done
3372-T was done
3374-T was done
3375-T was done
3376-T was done
3377-T was done
3382-T was done
3385-T was done
3386-T was done
3387-T was done
3388-T was done
3390-T w

4241-T was done
4242-T was done
4243-T was done
4245-T was done
4246-T was done
4248-T was done
4272-T was done
4274-T was done
4275-T was done
4282-T was done
4284-T was done
4286-T was done
4287-T was done
4288-T was done
4290-T was done
4291-T was done
4293-T was done
4295-T was done
4298-T was done
4299-T was done
4301-T was done
4304-T was done
4307-T was done
4308-T was done
4310-T was done
4312-T was done
4316-T was done
4317-T was done
4318-T was done
4319-T was done
4320-T was done
4320-S was done
4321-T was done
4323-T was done
4324-T was done
4326-T was done
4327-T was done
4331-T was done
4333-T was done
4334-T was done
4335-T was done
4336-T was done
4337-T was done
4341-T was done
4342-T was done
4343-T was done
4344-T was done
4345-T was done
4346-T was done
4347-T was done
4348-T was done
4350-T was done
4351-T was done
4355-T was done
4356-T was done
4361-T got IOError, added to errList
4362-T was done
4364-T was done
4365-T was done
4366-T was done
4367-T was done
436

5817-T was done
5819-T was done
5820-T was done
5821-T was done
5851-T was done
5852-T was done
5856-T was done
5857-T was done
5900-T was done
5901-T was done
5902-T was done
5902-S was done
5903-T was done
5905-T was done
5906-T was done
5907-T was done
5909-T was done
5911-T was done
5912-T was done
5915-T was done
5918-T was done
5921-T was done
5922-T was done
5923-T was done
5928-T was done
5929-T was done
5930-T was done
5932-T was done
5933-T was done
5935-T was done
5936-T was done
5938-T was done
5939-T was done
5940-T was done
5941-T was done
5942-T was done
5943-T was done
5945-T was done
5946-T was done
5947-T was done
5949-T was done
5950-T was done
5951-T was done
5952-T was done
5953-F was done
5955-T was done
5956-T was done
5957-T was done
5958-T was done
5959-T was done
5962-T was done
5964-T was done
5965-T was done
5966-T was done
5967-T got IOError, added to errList
5969-T was done
5970-T was done
5971-T was done
5973-T was done
5974-T was done
5975-T was done
597

KeyboardInterrupt: 

In [98]:
print(np.where(companyPd["StockSymbol"]=="6445-T"))

(array([2378]),)


In [100]:
for eachS in companyPd["StockSymbol"][2378:]:
    #print(each)
    tempURL = "http://k-db.com/stocks/" + eachS + "/5m/20170619?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + eachS + ".csv"
    time.sleep(1)
    try:
        eachFile.retrieve(tempURL, eachPath)
    except IOError:
        errList.append(eachS)
        print(eachS + " got IOError, added to errList")
    else:
        print(eachS + " was done")
    
    #sleepTime = random.randint(1,4)
    time.sleep(1)

6445-T was done
6448-T was done
6454-T was done
6455-T was done
6457-T was done
6458-T was done
6459-T was done
6460-T was done
6461-T was done
6462-T was done
6463-T was done
6464-T was done
6465-T was done
6466-T was done
6467-T was done
6469-T was done
6470-T was done
6471-T was done
6472-T was done
6473-T was done
6474-T was done
6478-T was done
6479-T was done
6480-T was done
6481-T was done
6482-T was done
6484-T got IOError, added to errList
6485-T was done
6486-T was done
6488-T was done
6489-T was done
6490-T was done
6492-T was done
6492-F was done
6493-T was done
6494-T was done
6495-T was done
6496-T was done
6497-T was done
6498-T was done
6501-T was done
6502-T was done
6503-T was done
6504-T was done
6504-F was done
6505-T was done
6506-T was done
6506-F was done
6507-T was done
6508-T was done
6513-T was done
6516-T was done
6517-T was done
6518-T was done
6531-T was done
6532-T was done
6533-T was done
6534-T was done
6535-T was done
6537-T was done
6537-F was done
653

7516-T was done
7517-T was done
7518-T was done
7519-T was done
7520-T was done
7521-T was done
7522-T was done
7523-T was done
7524-T was done
7525-T was done
7525-F was done
7527-T was done
7531-T was done
7532-T was done
7533-F was done
7537-T was done
7538-T was done
7539-T was done
7544-T was done
7545-T was done
7550-T was done
7551-T was done
7552-T was done
7554-T was done
7555-T was done
7559-T was done
7561-T was done
7562-T was done
7564-T was done
7565-T was done
7567-T was done
7570-T was done
7571-T was done
7575-T was done
7577-T was done
7578-T was done
7581-T was done
7585-T was done
7587-T was done
7590-T was done
7591-T was done
7593-T was done
7594-T was done
7595-T was done
7596-T was done
7597-T was done
7599-T was done
7600-T was done
7601-T was done
7602-T was done
7603-T was done
7604-T was done
7605-T was done
7606-T was done
7607-T was done
7608-T was done
7609-T was done
7610-T was done
7611-T was done
7612-T was done
7613-T was done
7614-T was done
7615-T w

8370-T was done
8374-T was done
8377-T was done
8377-S was done
8379-T was done
8381-T was done
8382-T was done
8383-T was done
8385-T was done
8386-T was done
8387-T was done
8388-T was done
8392-T was done
8392-F was done
8393-T was done
8393-F was done
8395-T was done
8395-F was done
8396-T was done
8396-F was done
8397-T was done
8397-F was done
8398-F was done
8399-T was done
8399-F was done
8410-T was done
8411-T was done
8416-T was done
8418-T was done
8421-T was done
8423-T was done
8424-T was done
8425-T was done
8439-T was done
8462-T was done
8473-T was done
8508-T was done
8511-T was done
8515-T was done
8518-T was done
8519-T was done
8521-T was done
8522-T was done
8524-T was done
8524-S was done
8527-T was done
8529-T was done
8530-T was done
8537-T was done
8540-F was done
8541-T was done
8542-T was done
8543-T was done
8544-T was done
8545-T was done
8550-T was done
8551-T was done
8554-F was done
8558-T was done
8559-F was done
8560-F was done
8562-T was done
8563-T w

9746-T was done
9747-T was done
9749-T was done
9753-T was done
9755-T was done
9757-T was done
9758-T was done
9759-T was done
9760-T was done
9761-T was done
9763-T was done
9764-T was done
9765-T was done
9766-T was done
9767-T was done
9768-T was done
9769-T was done
9776-T was done
9776-S was done
9778-T was done
9780-T was done
9782-T was done
9783-T was done
9787-T was done
9788-T was done
9790-T was done
9791-T was done
9792-T was done
9793-T was done
9795-T was done
9797-T was done
9799-T was done
9810-T was done
9812-T was done
9816-T was done
9818-T was done
9820-T got IOError, added to errList
9823-T was done
9824-T was done
9826-T was done
9827-T was done
9828-T was done
9829-T was done
9830-T was done
9831-T was done
9832-T was done
9835-T was done
9836-T was done
9837-T was done
9842-T was done
9843-T was done
9843-S was done
9845-T was done
9846-T was done
9849-T was done
9850-T was done
9852-T was done
9853-T was done
9854-T was done
9856-T was done
9857-T was done
986

In [104]:
print (errList)
print len(errList)

['1984-T', '1989-T', '1994-T', '2164-T', '2424-T', '2503-S', '2698-T', '2788-T', '2871-T', '2876-T', '2911-T', '3099-T', '3178-T', '3227-T', '3407-T', '3778-T', '3841-T', '3845-T', '3865-T', '3946-T', '4043-T', '4189-T', '4361-T', '5332-T', '5967-T', '6091-T', '6143-T', '6238-T', '6445-T', '6448-T', '6454-T', '6455-T', '6457-T', '6484-T', '6540-T', '7181-T', '7455-T', '7616-T', '7723-T', '7735-T', '7745-T', '7838-T', '7886-T', '7938-T', '8072-T', '8233-T', '8343-T', '8958-T', '8961-T', '8986-T', '9435-T', '9474-F', '9820-T', '9904-T']
54


### 不要用輕易用普通的 "=" ，這個只是參照，要用 copy()

In [108]:
errCrawlList = copy.copy(errList)
print (errCrawlList)


['1984-T', '1989-T', '1994-T', '2164-T', '2424-T', '2503-S', '2698-T', '2788-T', '2871-T', '2876-T', '2911-T', '3099-T', '3178-T', '3227-T', '3407-T', '3778-T', '3841-T', '3845-T', '3865-T', '3946-T', '4043-T', '4189-T', '4361-T', '5332-T', '5967-T', '6091-T', '6143-T', '6238-T', '6445-T', '6448-T', '6454-T', '6455-T', '6457-T', '6484-T', '6540-T', '7181-T', '7455-T', '7616-T', '7723-T', '7735-T', '7745-T', '7838-T', '7886-T', '7938-T', '8072-T', '8233-T', '8343-T', '8958-T', '8961-T', '8986-T', '9435-T', '9474-F', '9820-T', '9904-T']


In [111]:
while len(errCrawlList) > 0:
    target = errCrawlList[0]
        #print(each)
    tempURL = "http://k-db.com/stocks/" + target + "/5m/20170619?download=csv"
    eachFile = urllib.URLopener()
    eachPath = "./" + todayStr + "/" + target + ".csv"
    time.sleep(1)
    try:
        eachFile.retrieve(tempURL, eachPath)
    except IOError:
        #errList.append(eachS)
        print(target + " got IOError, added to errList")
        time.sleep(3)
        
    else:
        print(target + " was done")
        errCrawlList.remove(target)
    
    #sleepTime = random.randint(1,4)
    time.sleep(1)
    

1984-T was done
1989-T was done
1994-T was done
2164-T was done
2424-T was done
2503-S was done
2698-T was done
2788-T was done
2871-T was done
2876-T was done
2911-T was done
3099-T was done
3178-T was done
3227-T was done
3407-T was done
3778-T was done
3841-T was done
3845-T was done
3865-T was done
3946-T was done
4043-T was done
4189-T was done
4361-T was done
5332-T was done
5967-T was done
6091-T was done
6143-T was done
6238-T was done
6445-T was done
6448-T was done
6454-T was done
6455-T was done
6457-T got IOError, added to errList
6457-T got IOError, added to errList
6457-T was done
6484-T was done
6540-T was done
7181-T was done
7455-T was done
7616-T was done
7723-T was done
7735-T was done
7745-T was done
7838-T was done
7886-T was done
7938-T was done
8072-T got IOError, added to errList
8072-T was done
8233-T was done
8343-T was done
8958-T was done
8961-T was done
8986-T was done
9435-T was done
9474-F was done
9820-T was done
9904-T was done
